src/e2ee/sharedKeyManagement.ts

```diff
@@ -17,35 +17,37 @@ limitations under the License.
 import { useEffect, useMemo } from "react";
 
 import { useEnableE2EE } from "../settings/useSetting";
-import { useLocalStorage } from "../useLocalStorage";
+import { setLocalStorageItem, useLocalStorage } from "../useLocalStorage";
 import { useClient } from "../ClientContext";
 import { useUrlParams } from "../UrlParams";
 import { widget } from "../widget";
 
 export const getRoomSharedKeyLocalStorageKey = (roomId: string): string =>
   `room-shared-key-${roomId}`;
 
-const useInternalRoomSharedKey = (
-  roomId: string
-): [string | null, (value: string) => void] => {
+const useInternalRoomSharedKey = (roomId: string): string | null => {
   const key = useMemo(() => getRoomSharedKeyLocalStorageKey(roomId), [roomId]);
   const [e2eeEnabled] = useEnableE2EE();
-  const [roomSharedKey, setRoomSharedKey] = useLocalStorage(key);
+  const roomSharedKey = useLocalStorage(key)[0];
 
-  return [e2eeEnabled ? roomSharedKey : null, setRoomSharedKey];
+  return e2eeEnabled ? roomSharedKey : null;
 };
 
 const useKeyFromUrl = (roomId: string): string | null => {
   const urlParams = useUrlParams();
-  const [e2eeSharedKey, setE2EESharedKey] = useInternalRoomSharedKey(roomId);
+  const e2eeSharedKey = useInternalRoomSharedKey(roomId);
 
   useEffect(() => {
     if (!urlParams.password) return;
     if (urlParams.password === "") return;
     if (urlParams.password === e2eeSharedKey) return;
+    if (!urlParams.roomId) return;
```

Comment:
why do we not need to check:
```suggestion
    if (!urlParams.roomId) return;
    if (urlParams.roomId != roomId) return;
```
(or why do we not use the url roomId for `const e2eeSharedKey = useInternalRoomSharedKey(roomId);`)

Reply:
Mmm, I'm looking at this now and passing the room ID in as well is sort of confusing and it's not 100% clear what exactly it should be doing with it. I figured we would probably want to store the password in any case, as long as we store it with the correct room ID: otherwise if there's a race then we'd risk not picking up the password from the URL at all.

Reply:
OK, refactored a bit - I think it should make a bit more sense now.